cpp/src/io/parquet/page_enc.cu

```diff
@@ -1153,45 +1150,60 @@ inline __device__ uint8_t *cpw_put_fldh(uint8_t *p, int f, int cur, int t)
   }
 }
 
-#define CPW_BEGIN_STRUCT(hdr_start) \
-  {                                 \
-    uint8_t *p  = hdr_start;        \
-    int cur_fld = 0;
-
-#define CPW_FLD_STRUCT_BEGIN(f)                         \
-  p       = cpw_put_fldh(p, f, cur_fld, ST_FLD_STRUCT); \
-  cur_fld = 0;
-
-#define CPW_FLD_STRUCT_END(f) \
-  *p++    = 0;                \
-  cur_fld = f;
-
-#define CPW_FLD_INT32(f, v)                          \
-  p       = cpw_put_fldh(p, f, cur_fld, ST_FLD_I32); \
-  p       = cpw_put_int32(p, v);                     \
-  cur_fld = f;
-
-#define CPW_FLD_INT64(f, v)                          \
-  p       = cpw_put_fldh(p, f, cur_fld, ST_FLD_I64); \
-  p       = cpw_put_int64(p, v);                     \
-  cur_fld = f;
-
-#define CPW_FLD_BINARY(f, v, l)                   \
-  p = cpw_put_fldh(p, f, cur_fld, ST_FLD_BINARY); \
-  p = cpw_put_uint32(p, l);                       \
-  memcpy(p, v, l);                                \
-  p += l;                                         \
-  cur_fld = f;
-
-#define CPW_END_STRUCT(hdr_end) \
-  *p++    = 0;                  \
-  hdr_end = p;                  \
+struct cpw {
+  uint8_t *p;
+  int cur_fld;
+
+  inline __device__ cpw(uint8_t *hdr_start) : p(hdr_start), cur_fld(0) {}
+
+  inline __device__ void field_struct_begin(int f)
+  {
+    p       = cpw_put_fldh(p, f, cur_fld, ST_FLD_STRUCT);
+    cur_fld = 0;
+  }
+
+  inline __device__ void field_struct_end(int f)
+  {
+    *p++    = 0;
+    cur_fld = f;
+  }
+
+  template <typename T>
+  inline __device__ void field_int32(int f, T v)
```

Comment:
Please replace the single-letter variables with meaningful names.
Also avoid shortened versions of words, like `hdr` and `fld`. (I think _ptr is fine)